client/error.py

```diff
@@ -240,6 +265,70 @@ def to_text(self) -> str:
         column = click.style(str(self.column), fg="yellow")
         return f"{path}:{line}:{column} {self.description}"
 
+    def to_sarif(self) -> Dict[str, Any]:
+        return {
+            "ruleId": "model-verification-error",
+            "level": "error",
+            "message": {"text": self.description},
+            "locations": [
+                {
+                    "physicalLocation": {
+                        "artifactLocation": {
+                            "uri": str(self.path),
+                        },
+                        "region": {
+                            "startLine": self.line + 1,
+                            "startColumn": self.column + 1,
+                            "endLine": self.stop_line + 1,
+                            "endColumn": self.stop_column + 1,
+                        },
+                    },
+                },
+            ],
+        }
+
+
+def errors_to_sarif(
+    errors: Union[
+        Sequence[Error], Sequence[LegacyError], Sequence[ModelVerificationError]
+    ]
+) -> Dict[str, Any]:
+    return {
+        "version": "2.1.0",
+        "$schema": "http://json.schemastore.org/sarif-2.1.0-rtm.4.json",
+        "runs": [
+            {
+                "tool": {
+                    "driver": {
+                        "name": "Pyre",
+                        "informationUri": "https://www.pyre-check.org",
+                        "rules": [
```

Comment:
Not sure if this is currently possible, but it would be cool if we could list [specific codes for type errors](https://pyre-check.org/docs/errors/#error-codes)

https://github.com/facebook/pyre-check/blob/bfa69fa7198215e07fcdd20d6e81fc5a03dc0253/source/analysis/analysisError.ml#L422-L557

Reply:
@0xedward nice suggestion! Made the required changes. (Teaser: https://github.com/abishekvashok/pyre-check/security/code-scanning/3?query=ref%3Arefs%2Fpull%2F4%2Fmerge)